### PR TITLE
add the "role" command group

### PR DESCRIPTION
**Pull request by GlaZedBelmont**
DON'T MERGE YET

### CODE REVIEW

**Comment by GlaZedBelmont** [comment]
Merge ready!

File: cogs/mod.py [hunk 1]
```diff
@@ -654,6 +654,54 @@ async def approve(self, ctx, alias: str, invite: discord.Invite, times: int = 1)
...
+
+    @is_staff("HalfOP")
+    @role.command(name='give')
+    async def give_role(self, ctx, member: FetchMember, *, role: str):
+        if member is None:
+            return await ctx.send("Please include a user to add a role to.")
+
+        selected_role = [x for x in self.role_list if role in x]
+
+        if len(selected_role) > 1:
```
Comment (FrozenChen) [inline cogs/mod.py:683]: what if there was no match and selected_role is a empty list? then you are getting a index error when trying to add the role

File: cogs/mod.py [hunk 2]
```diff
@@ -654,6 +654,54 @@ async def approve(self, ctx, alias: str, invite: discord.Invite, times: int = 1)
...
+
+        if len(selected_role) > 1:      
+            return await ctx.send("More than a role were selected. Stopping...")
+
+        await member.add_roles(self.bot.roles[selected_role[0]])
+        await ctx.send(f"{member.mention} has been given the ``{selected_role}`` role.")
+
+
+    @is_staff("HalfOP")
+    @role.command(name='take', aliases=['remove', 'delete', 'del'])
```
Comment (FrozenChen) [inline cogs/mod.py:691]: aliases are too generic

File: cogs/mod.py [hunk 3]
```diff
@@ -654,6 +654,54 @@ async def approve(self, ctx, alias: str, invite: discord.Invite, times: int = 1)
         await self.bot.invitefilter.add(code=code, name=invite.guild.name, alias=alias, uses=times)
         await ctx.send(f"Approved an invite to {invite.guild}({code}) for posting {times} times")
         await self.bot.channels['mod-logs'].send(f"⭕ **Approved**: {ctx.author.mention} approved server {invite.guild}({code}) to be posted {times} times")
+                       
+
```
Comment (FrozenChen) [inline cogs/mod.py:658]: Only one empty line between commands

**Review by FrozenChen — CHANGES_REQUESTED** [review]

File: cogs/mod.py [hunk 4]
```diff
@@ -653,7 +653,50 @@ async def approve(self, ctx, alias: str, invite: discord.Invite, times: int = 1)
...
+    @is_staff("HalfOP")
+    @role.command(name='give')
+    async def give_role(self, ctx, member: SafeMember, *, role: str):
+
+        selected_role = [x for x in self.role_list if role in x]
+                       
+        if len(selected_role) > 0:
+            if len(selected_role) > 1:      
+                return await ctx.send("More than a role were selected. Stopping...")
+            await member.add_roles(self.bot.roles[selected_role[0]])
```
Comment (bleck9999) [inline cogs/mod.py:682]: There's no else here, it doesn't stop if there's more than 1 selected role

**Pull request closed without merging**